### PR TITLE
#385 arcsi analytics

### PR DESCRIPTION
**Pull request by baxgas**
Reworked the download to trigger the gtag and then open the link

### CODE REVIEW

File: wp/wp-content/themes/lahma_maker/js/scripts.js [hunk 1]
```diff
@@ -100,6 +100,15 @@ function arcsiStop(episodeName) {
     });
 }
 
+function arcsiDownload(episodeName, downloadLink) {
+    gtag('event', 'Arcsi download', {
+        'event_category': episodeName,
+        'event_label': 'Download name',
```
Comment (gammaw) [inline wp/wp-content/themes/lahma_maker/js/scripts.js:106]: Shouldn't `'Download name'` be `downloadLink` (the input parameter of the function)?

File: wp/wp-content/themes/lahma_maker/js/scripts.js [hunk 2]
```diff
@@ -100,6 +100,15 @@ function arcsiStop(episodeName) {
...
+        'value': 1,
+    });
```
Comment (gammaw) [inline wp/wp-content/themes/lahma_maker/js/scripts.js:108]: Why don't make it consistent with the `Radio play` event (category = show title)? See: https://github.com/mmmnmnm/lahmacun/blob/1d04ecd14873e7b5896419c778d157af08cd5154/wp/wp-content/themes/lahma_maker/assets/playerfiles/radio_player.vue#L326-L330

Also, we could use the `event_label` for the episode name, don't we? Not only here, for radio play events too (we never make sure of `Play state`).

File: wp/wp-content/themes/lahma_maker/template-parts/content-arcsi.php [hunk 3]
```diff
@@ -110,10 +116,10 @@ function sortAssociativeArrayByKey($array, $key, $direction){
         <h4 class="episode-name"><?php echo $showname; ?></h4>   
         <p><?php echo $showdescription; ?></p> 
         <div id="arcsi-audio-<?php echo $showid; ?>" class="arcsicontrols">
-            <a class="arcsibutton arcsidown" href="<?php echo $server; ?>/arcsi/item/<?php echo $showid; ?>/download" target="_blank">
+            <a class="arcsibutton arcsidown avoidAjax" href="#" data-href="<?php echo $server; ?>/arcsi/item/<?php echo $showid; ?>/download" title="<?php echo $fullTitle; ?>">
```
Comment (gammaw) [inline wp/wp-content/themes/lahma_maker/template-parts/content-arcsi.php:119]: What's the reason of the change? I tried both and didn't notice any difference (download opens a new tab, which closes immediately).

File: wp/wp-content/themes/lahma_maker/js/scripts.js (hunk 1 above)
Comment (baxgas) [reply wp/wp-content/themes/lahma_maker/js/scripts.js:106]: No, 'Download name' is the label that is sent to gtag, downloadLink is the link to the show itself
line 109
`    window.open(downloadLink)
`

File: wp/wp-content/themes/lahma_maker/js/scripts.js (hunk 2 above)
Comment (baxgas) [reply wp/wp-content/themes/lahma_maker/js/scripts.js:108]: ?? don't understand the comment. the download title is the same as the play title, we already agreed upon last time. you sent the radio stream title, which we get from different source
in content-arcsi.php
```
$fullTitle = $showtitle . ' | ' . $showname;
title="<?php echo $fullTitle; ?>
```

File: wp/wp-content/themes/lahma_maker/template-parts/content-arcsi.php (hunk 3 above)
Comment (baxgas) [reply wp/wp-content/themes/lahma_maker/template-parts/content-arcsi.php:119]: yes, this is how it works only, if it is a href link, the browser prioritize it and doesn't send the gtag, because it already opens a new window.
so the process how it works is like this arcsidown triggers the `arcsiDownload()` function and sends in the title for gtag and the data-href link for the download. the function sends the gtag, and then opens the download link

File: wp/wp-content/themes/lahma_maker/js/scripts.js (hunk 1 above)
Comment (gammaw) [reply wp/wp-content/themes/lahma_maker/js/scripts.js:106]: Oh I see.

File: wp/wp-content/themes/lahma_maker/template-parts/content-arcsi.php (hunk 3 above)
Comment (gammaw) [reply wp/wp-content/themes/lahma_maker/template-parts/content-arcsi.php:119]: I see. nice!

**Review by pvj — APPROVED** [review]
Sorry for the messy first implementation w/ "Play state" 

That's my bad. 

This has been well reimagined thanks to you both!

File: wp/wp-content/themes/lahma_maker/js/scripts.js [hunk 4]
```diff
@@ -300,14 +303,16 @@ $( document ).ready(function() {
 
     // Arcsi Download
     $(document).on("click", ".arcsidown", function(e) {
-        audioTitle = $(this).attr('title') 
-        arcsiDownload(audioTitle)
+        e.preventDefault();
+        showTitle = $(this).data('showtitle')
```
Comment (gammaw) [inline wp/wp-content/themes/lahma_maker/js/scripts.js:307]: Note: these are data-* accesses (see: https://api.jquery.com/data/) and compare with the data-* fields e.g. data-showtitle in content-arcsi.php
